lib/nrf_modem_lib/trace_backends/uart/Kconfig

```diff
@@ -8,18 +8,37 @@
 choice NRF_MODEM_LIB_TRACE_BACKEND
 
 config NRF_MODEM_LIB_TRACE_BACKEND_UART
-	bool "UARTE1"
-	# Modem tracing over UART use the UARTE1 as a dedicated peripheral.
-	# This enables the UARTE1 peripheral and includes nrfx UARTE driver.
-	select NRFX_UARTE1
+	bool "UART"
 
 if NRF_MODEM_LIB_TRACE_BACKEND_UART
 
+choice NRF_MODEM_LIB_TRACE_BACKEND_UART_VERSION
+	prompt "UART trace backend version"
+	default NRF_MODEM_LIB_TRACE_BACKEND_UART_ASYNC
+
+config NRF_MODEM_LIB_TRACE_BACKEND_UART_ASYNC
+	bool "UARTE1 ASYNC"
+	select NRFX_UARTE1
+	select DEPRECATED
+	help
+	  Enable synchronous processing for UARTE1 trace backend.
```

Comment:
Is the help text or name here wrong?
Does it enable synchronous or asynchronous processing of the traces?